### PR TITLE
Add retry limit for temporary errors in waiters

### PR DESCRIPTION
**Pull request by vicentepinto98**

### CODE REVIEW

File: core/wait/wait_test.go [hunk 1]
```diff
@@ -59,14 +61,46 @@ func TestWait_SetThrottle(t *testing.T) {
 	}
 }
 
+func TestWait_SetSleepBeforeWait(t *testing.T) {
```
Comment (hcsa73) [inline core/wait/wait_test.go:64]: Why not just call this SetSleepBeforeWait?

**Review by hcsa73 — DISMISSED** [review]

File: core/wait/wait.go [hunk 2]
```diff
@@ -5,26 +5,31 @@ import (
...
 type Handler struct {
-	fn              WaitFn
-	sleepBeforeWait time.Duration
-	throttle        time.Duration
-	timeout         time.Duration
+	fn                WaitFn
+	sleepBeforeWait   time.Duration
+	throttle          time.Duration
+	timeout           time.Duration
+	retryLimitTempErr int
```
Comment (joaopalet) [inline core/wait/wait.go:22]: I would flip the name of this variable to be `tempErrRetryLimit`

File: core/wait/wait.go [hunk 3]
```diff
@@ -79,3 +92,22 @@ func (w *Handler) WaitWithContext(ctx context.Context) (res interface{}, err err
 		}
 	}
 }
+
+func handleError(retryTempErrorCounter, retryLimitTempErr int, err error) (int, error) {
```
Comment (joaopalet) [inline core/wait/wait.go:96]: Maybe we could add `(w *Handler)` as a receiver here so we don't have to pass down `retryLimitTempErr`

File: core/wait/wait_test.go [hunk 4]
```diff
@@ -146,3 +185,63 @@ func TestWait_SetTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleError(t *testing.T) {
+	tests := []struct {
+		desc               string
+		reqErr             error
+		tempErroRetryLimit int
```
Comment (joaopalet) [inline core/wait/wait_test.go:193]: Small typo
```suggestion
		tempErrRetryLimit int
```

**Review by joaopalet — APPROVED** [review]

**Review by joaopalet — APPROVED** [review]

**Pull request merged**